packages/core/src/subscribe.ts

```diff
@@ -0,0 +1,56 @@
+import { execute as defaultExecute, createSourceEventStream } from 'graphql';
+import isAsyncIterable from 'graphql/jsutils/isAsyncIterable';
+import mapAsyncIterator from 'graphql/subscription/mapAsyncIterator';
+import { makeSubscribe } from './util';
+
+/**
+ * This is a almost identical port from graphql-js subscribe.
+ * The only difference is that a custom `execute` function can be injected as an additional argument.
+ */
+export const subscribe = makeSubscribe(async args => {
```

Comment:
Maybe this can live inside the plugin, and then you can do `setSubscribeFn` and replace the function?

Reply:
Would prefer it to be a core or custom plugin?

Reply:
Also, I am not sure how we should treat the subscribe function typings. Let's say we only use this custom subscribe function via a plugin. Should we still use the typings I defined in packages/types/src/index.ts for the subscribe function users can provide? How would the hooks API work for custom plugins that want to hook into the subscribe execution phase?

Reply:
Not sure about the types, but you are correct, it needs to be adjusted somehow. The plugin for now can assume that it replaces the function and then call it with the custom params? Not 100% sure actually. 

I think `useContextPerSubscriptionEvent` should be a separate plugin package because it adds a specific functionality that isn't needed for most users. So we can have that as a separate plugin, and then we can have the custom subscribe method there (and you can use `setSubscribeFn` to override it)

Reply:
I removed the new hooks and instead moved stuff into a separate plugin. I actually think this is badly needed by anyone using DataLoader and subscriptions.

Maybe we can move it back to the core once graphql-js allows providing a custom execute to subscribe. For now, all the types are aligned.

Reply:
As a follow up we maybe can also add a generic `subscribeExecute` that allows hooking into all phases of the execute call.